packages/odyssey-react-mui/src/ui-shell/NarrowUiShellContent.tsx

```diff
@@ -0,0 +1,383 @@
+    "banners"
+    "top-nav"
+    "app-content"
+  `,
+  gridTemplateColumns: "1fr",
+  gridTemplateRows: "auto auto 1fr",
+  height: "100vh",
+  width: "100vw",
+}));
+
+const StyledLogoContainer = styled("div")({
+  "svg, img": {
+    maxHeight: "100%",
+    width: "auto",
+    maxWidth: "100%",
+  },
+});
+
+const StyledTopNav = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" &&
+    prop !== "isContentScrolled" &&
+    prop !== "topNavBackgroundColor",
+})<{
+  isContentScrolled?: boolean;
+  odysseyDesignTokens: DesignTokens;
+  topNavBackgroundColor?: UiShellColors["topNavBackgroundColor"];
+}>(({ odysseyDesignTokens, isContentScrolled, topNavBackgroundColor }) => ({
+  alignItems: "stretch",
+  backgroundColor: topNavBackgroundColor,
+  boxShadow: isContentScrolled ? odysseyDesignTokens.DepthMedium : undefined,
+  clipPath: "inset(0 0 -100vh 0)",
+  display: "flex",
+  flexDirection: "column",
+  gridArea: "top-nav",
+  height: "100%",
+  justifyContent: "center",
+  position: "relative",
+  transition: `box-shadow ${odysseyDesignTokens.TransitionDurationMain} ${odysseyDesignTokens.TransitionTimingMain}`,
+  zIndex: 100,
+}));
+
+const StyledTopNavMenu = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" && prop !== "topNavBackgroundColor",
+})<{
+  odysseyDesignTokens: DesignTokens;
+  topNavBackgroundColor?: UiShellColors["topNavBackgroundColor"];
+}>(({ odysseyDesignTokens, topNavBackgroundColor }) => ({
+  alignItems: "center",
+  display: "flex",
+  gap: odysseyDesignTokens.Spacing3,
+  height: "100%",
+  justifyContent: "space-between",
+  maxHeight: TOP_NAV_HEIGHT,
+  minHeight: TOP_NAV_HEIGHT,
+  paddingBlock: odysseyDesignTokens.Spacing2,
+  paddingInline: odysseyDesignTokens.Spacing3,
+  transition: `box-shadow ${odysseyDesignTokens.TransitionDurationMain} ${odysseyDesignTokens.TransitionTimingMain}`,
+
+  ...(topNavBackgroundColor === odysseyDesignTokens.HueNeutralWhite &&
+    ({
+      borderBottomColor: odysseyDesignTokens.HueNeutral100,
+      borderBottomStyle: odysseyDesignTokens.BorderStyleMain,
+      borderBottomWidth: odysseyDesignTokens.BorderWidthMain,
+    } as CSSProperties)),
+}));
+
+const StyledTopNavSearch = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  paddingBlock: odysseyDesignTokens.Spacing2,
+  paddingInline: odysseyDesignTokens.Spacing3,
+}));
+
+export type NarrowUiShellContentProps = Pick<HtmlProps, "testId"> &
+  Pick<UiShellNavComponentProps, "sideNavProps" | "topNavProps"> &
+  UnifiedUiShellContentProps;
+
+const NarrowUiShellContent = ({
+  appContainerElement,
+  appContainerScrollingMode,
+  hasStandardAppContentPadding = true,
+  initialVisibleSections = ["TopNav", "SideNav", "AppSwitcher"],
+  onError = console.error,
+  optionalComponents,
+  sideNavProps,
+  topNavProps,
+}: NarrowUiShellContentProps) => {
```

Comment:
This component is enormous. Arguably, some of it should be split out like the `TopNav` part. The rest is related to this component's rendering.